R/auditCommonFunctions.R

```diff
@@ -1580,682 +1406,629 @@
-  if (options[["separateKnownAndUnknownMisstatement"]])
+.jfaPlotSampleSizes <- function(options, parentOptions, parentState, parentContainer, jaspResults, ready, positionInContainer) {
+  if (!options[["plotSampleSizes"]] || !options[["materiality_test"]] || options[["separateMisstatement"]]) {
     return()
-  
-  if (!options[["priorConstructionMethod"]] %in% c("none", "arm", "sample", "factor"))
-    return()
-  
+  }
+
   .jfaFigureNumberUpdate(jaspResults)
-  
+
   if (is.null(parentContainer[["decisionPlot"]])) {
-    
-    collection <- createJaspContainer(gettext("Comparison of Required Sample Sizes"))
-    collection$dependOn(options = c("decisionPlot"))
+    collection <- createJaspContainer(gettext("Minimum Sample Sizes"))
+    collection$dependOn(options = c("plotSampleSizes"))
     collection$position <- positionInContainer
-    
+
     parentContainer[["decisionPlot"]] <- collection
-    
-    if (!ready || parentContainer$getError())
+
+    if (!ready || parentContainer$getError()) {
       return()
-    
-    confidence <- options[["confidence"]]
-    if (!options[["bayesianAnalysis"]])
-      confidence <- 1 - .jfaAuditRiskModelCalculation(options)
-    
-    minPrecision <- NULL
-    if (options[["minimumPrecision"]])
-      minPrecision <- parentOptions[["minimumPrecision"]]
-    
-    performanceMateriality <- NULL
-    if (options[["performanceMateriality"]])
-      performanceMateriality <- parentOptions[["materiality"]]
-    
-    N <- parentOptions[["populationSize"]]
-    if (options[["workflow"]] && options[["monetaryVariable"]] != "")
-      N <- ceiling(parentOptions[["populationValue"]])
-    if (!options[["workflow"]] && options[["populationValue"]] > 0)
-      N <- ceiling(parentOptions[["populationValue"]])
-    
-    if (options[["bayesianAnalysis"]])
-      dist <- base::switch(parentOptions[["likelihood"]], "binomial" = "Beta", "poisson" = "Gamma", "hypergeometric" = "Beta-binomial")
-    if (!options[["bayesianAnalysis"]])
-      dist <- base::switch(parentOptions[["likelihood"]], "binomial" = "Binomial", "poisson" = "Poisson", "hypergeometric" = "Hypergeometric")
-    
+    }
+
+    confidence <- if (options[["bayesian"]]) options[["conf_level"]] else 1 - .jfaAuditRiskModelCalculation(options)
+    min_precision <- if (options[["min_precision_test"]]) options[["min_precision_rel_val"]] else NULL
+    materiality <- if (options[["materiality_test"]]) parentState[["materiality"]] else NULL
+
+    N <- parentState[["N.units"]]
+    if (options[["workflow"]] && options[["values"]] != "") {
+      N <- ceiling(parentState[["N.units"]])
+    }
+    if (!options[["workflow"]] && parentState[["N.units"]] > 0) {
+      N <- ceiling(parentState[["N.units"]])
+    }
+
+    if (options[["bayesian"]]) {
+      dist <- switch(options[["likelihood"]],
+        "binomial" = "Beta",
+        "poisson" = "Gamma",
+        "hypergeometric" = "Beta-binomial"
+      )
+    } else {
+      dist <- switch(options[["likelihood"]],
```

Comment:
distName (?) Here you're creating the name of the distribution to be used in the plot title.